### ISSUE TITLE
Example about different learning rate for different layers

### ISSUE THREAD

**Issue by sxjscience** [google/flax#756]
First of all, thanks for writing this amazing package! I'm recently trying out flax linen and find it's really easy to use. For my use case, I'm going to implement the layerwise decay learning in the BERT finetuning. Basically, I will need to assign smaller LR to layers that are closer to the input, as described here https://github.com/google-research/electra/blob/f93f3f81cdc13435dd3e85766852d00ff3e00ab5/model/optimization.py#L181-L193.

Is there any example that have implemented this? (I tried to find examples in the linen_example folder but haven't found one. May be I've missed some examples).

**Comment by andsteing** [comment]
Hi Xingjian

Did you have a look at the `MultiOptimizer`?

https://github.com/google/flax/blob/e9ccc3945e2085b45504306f0b01f9065067a742/flax/optim/base.py#L395-L417

This lets you specify different optimizers (or the same optimizer with different learning rates) for different parts of your model.

I don't think we currently use it in any of the examples, but that we might add it eventually.

**Comment by sxjscience** [comment]
No, I haven't found this class but thanks for offering this pointer!

**Comment by sxjscience** [comment]
FYI @Cli212

**Comment by Cli212** [comment]
> FYI @Cli212

Thanks for mentioning, this is very helpful to me.

**Comment by andsteing** [comment]
Closing the issue since the question seems answered. Feel free to re-open if that's not the case...
